src/java.base/share/classes/sun/net/www/protocol/jrt/JavaRuntimeURLConnection.java

```diff
@@ -45,99 +44,79 @@
  */
 public class JavaRuntimeURLConnection extends URLConnection {
 
-    // ImageReader to access resources in jimage
-    private static final ImageReader reader = ImageReaderFactory.getImageReader();
-
-    // the module and resource name in the URL
+    // ImageReader to access resources in jimage (never null).
+    private static final ImageReader READER = ImageReaderFactory.getImageReader();
+
+    // The module and resource name in the URL ("jrt:/<module-name>/<resource-name>").
+    //
+    // It is important to note that all of this information comes from the given
+    // URL's path part, and there's no requirement for there to be distinct rules
+    // about percent encoding, and it is likely that any differences between how
+    // module names and resource names are treated is unintentional. The rules
+    // about percent encoding may well be tightened up in the future.
+    //
+    // The module name is not percent-decoded, and can be empty.
     private final String module;
+    // The resource name permits UTF-8 percent encoding of non-ASCII characters.
     private final String name;
 
-    // the Resource when connected
-    private volatile Resource resource;
+    // The resource node (when connected).
+    private volatile Node resource;
 
     JavaRuntimeURLConnection(URL url) throws IOException {
         super(url);
-        String path = url.getPath();
-        if (path.isEmpty() || path.charAt(0) != '/')
+        // TODO: Allow percent encoding in module names.
+        // TODO: Consider rejecting URLs with fragments, queries or authority.
+        String urlPath = url.getPath();
+        if (urlPath.isEmpty() || urlPath.charAt(0) != '/') {
             throw new MalformedURLException(url + " missing path or /");
-        if (path.length() == 1) {
-            this.module = null;
+        }
+        int pathSep = urlPath.indexOf('/', 1);
+        if (pathSep == -1) {
+            // No trailing resource path. This can never "connect" or return a
+            // resource, but might be useful as a representation to pass around.
+            // The module name *can* be empty here (e.g. "jrt:/") but not null.
```

Comment:
If the URL scheme from JEP 220 goes into the first comment then it will be clear that the resource is optionally. This will allow the comment here to be reduced.